### PR TITLE
Updated function names in ovl_En_Bom & added documentation from OoT

### PR DESCRIPTION
**Pull request by swinginman**
Function names, variables, & comments were all copied from:
[https://github.com/zeldaret/oot/blob/master/src/overlays/actors/ovl_En_Bom/z_en_bom.c](url)

### CODE REVIEW

File: src/overlays/actors/ovl_En_Bom/z_en_bom.c [hunk 1]
```diff
@@ -542,7 +552,7 @@ void EnBom_Update(Actor* thisx, PlayState* play) {
                 thisx->params = ENBOM_1;
                 this->timer = 10;
                 thisx->flags |= (ACTOR_FLAG_20 | ACTOR_FLAG_100000);
-                this->actionFunc = func_808715B8;
+                this->actionFunc = EnBom_Explode;
```
Comment (isghj5) [inline src/overlays/actors/ovl_En_Bom/z_en_bom.c:555]: Considering it explodes right after having its params set to ENBOM_1, and it checks for that before setting the AT attack collider, I suspect ENBOM_1 means its exploding, can probably be named.

**Review by isghj5 — COMMENTED** [review]
Looks pretty good, I would add that since function `func_80872BC0` seems to be the only function with Keg assets, you could probably name it to `EnBom_DrawKeg`

File: src/overlays/actors/ovl_En_Bom/z_en_bom.c (hunk 1 above)
Comment (swinginman) [reply src/overlays/actors/ovl_En_Bom/z_en_bom.c:555]: Oh yeah in OoT ENBOM_0 is called BOMB_BODY, and ENBOM_1 is called BOMB_EXPLOSION.

Comment (swinginman) [reply src/overlays/actors/ovl_En_Bom/z_en_bom.c]: Added in latest commit.

File: src/overlays/actors/ovl_En_Bom/z_en_bom.c [hunk 2]
```diff
@@ -494,22 +499,27 @@ void EnBom_Update(Actor* thisx, PlayState* play) {
...
 
-            sp74.y = 0.2f;
-            sp80 = thisx->world.pos;
-            sp80.y += 10.0f;
+            dustAccel.y = 0.2f;
+            effPos = thisx->world.pos;
+            effPos.y += 10.0f;
 
+            // double bomb flash speed and adjust red color at certain times during the countdown
             if ((this->timer == sp60->x) || (this->timer == sp60->y) || (this->timer == 3)) {
```
[the remark this answers is not in the document]
Comment (swinginman) [reply src/overlays/actors/ovl_En_Bom/z_en_bom.c:523]: The values are all hard coded in OoT, so I don't know what that should be called either.
But I changed func_80872BC0 to EnBom_DrawKeg in the latest commit.

**Comment by swinginman** [comment]
Committed without running warning checks ([d8b800e](https://github.com/zeldaret/mm/pull/1170/commits/d8b800e1683f4d5a54a9e1ecad1d627177ed85a0))
ENBOM_0 was being referenced in ovl_En_Famos and ovl_En_Rat actors.
Commit [19b4908](https://github.com/zeldaret/mm/pull/1170/commits/19b49084c5b214c21ea5862ae1a16e5a57eb7dbb) has fixed this.

File: src/overlays/actors/ovl_En_Bom/z_en_bom.h [hunk 3]
```diff
@@ -12,8 +12,8 @@ typedef void (*EnBomActionFunc)(struct EnBom*, PlayState*);
 #define ENBOM_GETZ_FF00(thisx) (((thisx)->shape.rot.z & 0xFF00) >> 8)
 
 enum {
-    /* 0 */ ENBOM_0,
-    /* 1 */ ENBOM_1,
+    /* 0 */ BOMB_BODY,
+    /* 1 */ BOMB_EXPLOSION,
 };
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Bom/z_en_bom.h:17]:
```c
typedef enum {
    /* 0 */ BOMB_BODY,
    /* 1 */ BOMB_EXPLOSION,
} BombType;
```
When enums are documented, they should be typedef'd with a name

File: src/overlays/actors/ovl_En_Bom/z_en_bom.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (engineer124) [reply src/overlays/actors/ovl_En_Bom/z_en_bom.c:523]: Yeah, this is a bit weird. I don't think `Vec3us*` is the appropriate type for `sp60`/`D_80872ED4`, I think I'd make a custom struct with `timer1/2/3`, but I wouldn't worry too much if you want to keep your first PR simple

[the remark this answers is not in the document]
Comment (swinginman) [reply src/overlays/actors/ovl_En_Bom/z_en_bom.c:523]: Yeah lets keep this one simple 👍

**Review by AngheloAlf — APPROVED** [review]
Nice work!

**Review by engineer124 — APPROVED** [review]

**Review by tom-overton — APPROVED** [review]

**Comment by EllipticEllipsis** [comment]
This all looks good to me. Could you agree to share your contributions under CC0 here: https://github.com/louist103/mm-licensing/issues/1 and then we can get this merged?

**Comment by swinginman** [comment]
>

**Comment by swinginman** [comment]
> This all looks good to me. Could you agree to share your contributions under CC0 here: [louist103/mm-licensing#1](https://github.com/louist103/mm-licensing/issues/1) and then we can get this merged?

Sorry I missed this, and accidentally closed the issue.

I have agreed to the conditions under CC0, is there anything else you need from me ?

**Comment by AngheloAlf** [comment]
Great! The only remaining thing would be to fix the conflicts and ensure an OK builds. After that we can merge this PR c:

**Pull request closed without merging**